tests/test_enterprise/api/test_views.py

```diff
@@ -220,33 +185,28 @@ def test_enterprise_learner_entitlements(
                 (this includes enforcing data sharing consent at login and at enrollment) and enterprise learner
                  does not consent to share data.
             2. Full list of entitlements for all other cases.
-        Arguments:
-            enable_data_sharing_consent (bool): True if enterprise customer enables data sharing consent,
-                False it does not.
-            enforce_data_sharing_consent (str): string for the location at which enterprise customer enforces
-                data sharing consent, possible values are 'at_enrollment' and 'externally_managed'.
-            learner_consent_state (str): string containing the state of learner consent on data sharing,
-                possible values are 'not_set', 'enabled' and 'disabled'.
-            entitlements (list): A list of integers pointing to voucher ids generated in E-Commerce CAT tool.
-            expected_json (dict): A dict with structure and values corresponding to
-                the expected json from API endpoint.
         """
-        user_id = 1
+        user_id = self.user.id + 1
         enterprise_customer = factories.EnterpriseCustomerFactory(
             enable_data_sharing_consent=enable_data_sharing_consent,
             enforce_data_sharing_consent=enforce_data_sharing_consent,
         )
-        factories.UserDataSharingConsentAuditFactory(
-            user__id=user_id,
-            user__enterprise_customer=enterprise_customer,
-            state=learner_consent_state,
+        user = factories.UserFactory(id=user_id)
+        ecu = factories.EnterpriseCustomerUserFactory(
+            user_id=user.id,
+            enterprise_customer=enterprise_customer,
+        )
+        factories.DataSharingConsentFactory(
```

Comment:
Is this still doing something meaningful? Why aren't we passing a course ID (which should be required for this model), and why is the result working without a specific course ID? I might be missing something here.

Reply:
The endpoint's function attempts to [get entitlements](https://github.com/edx/edx-enterprise/blob/master/enterprise/api/v1/views.py#L220) which then only [looks to see](https://github.com/edx/edx-enterprise/blob/master/enterprise/models.py#L417) if any DSC records exist between the learner and the customer, and if one does and DSC is enforced, the logic says consent isn't required.

I don't know if this is how DSC requirements for entitlements should actually be working, so please let me know and we'd have to fix that fast.

Reply:
@douglashall, thoughts? I think we might have missed an edge case here; I think the appropriate fix would be to mark consent as required on the entitlements based _solely_ on the Enterprise setting, and then put logic in ecom to check for actual consent on the relevant courses. Or, you could even modify the existing consent views to redirect the user back to the success URL if consent straight-up isn't required.

Reply:
Is this something that would break our current coupon-voucher-based implementation of auto-discounts for enterprise learners? If so, we should fix it. Even though eventually, I think the entitlements endpoint can be removed entirely once we migrate over to using Enterprise Offers for the auto-enterprise-discount.

Reply:
@douglashall I think it might. @UmanShahzad, see my thoughts on the implementation above.

Reply:
@haikuginger @douglashall I will look into this, but I'm a little time-strapped with the programs landing page & the endpoint stuff. This issue was of course quite unexpected. I might have to ask @itsjeyd to help if I realize this may be a little too much for me alone right now. I'll let you guys know by tomorrow morning.

Reply:
See https://openedx.atlassian.net/browse/ENT-644?focusedCommentId=286220&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-286220 for further discussion on this.